### DISCUSSION TITLE
blade @checked() - It's possible to use?

### DISCUSSION THREAD

**Discussion by helderneves91** [livewire/flux#259] [Feature Requests] [answered]
Hi,

Lets take this code as example:
```
<flux:radio.group wire:model="priority_id">
    @foreach($this->priorities as $priority)
        <flux:radio
                value="{{ $priority->id }}"
                label="{{ $priority->name }}"
                description="{{ $priority->description }}"
                @checked((int)$exam->priority_id === (int)$priority->id))
        />
    @endforeach
</flux:radio.group>
```

This code doesn't show the radio inputs because of `@checked()`.

There is any way to do this? I really don't want to control the checked attribute with an if else 😱

Thanks,
Helder

**Comment by jeffchown** [comment, marked as answer]
Is it not possible to use `:checked="(int)$exam->priority_id === (int)$priority->id"`?

**Reply by helderneves91** [reply]
We can, but if we wire:model the group, the condition won't work.
Check https://github.com/livewire/flux/issues/260

**Reply by jeffchown** [reply]
I see.   I understand re: `disabled`.

As for `selected`, if you put a `wire:model` on the group, the value of the `wire:model` property should be responsible for indicating which radio button is initially selected, should it not?  Otherwise, the `wire:model` property and the `@selected` condition(s) could be in conflict.  Therefore, when would the `selected` property be necessary?

**Reply by helderneves91** [reply]
It's not initially selected, but should be since the option is in the loop. `:checked` only works if I don't wire It.
As a test I placed a select input to do exactly the same , and `:select` condition works.

**Reply by jeffchown** [reply]
In my case, whenever I `wire:model` an input (checkbox, radio, etc.), in my Livewire component I set the property to my desired default/starting value (if no default, then set to '').  I find using both `wire:model` and `:selected` to be confusing.

**Reply by helderneves91** [reply]
Facts, I do it to. In component I set the `$this->priority_id` to the actual record value, this should ensure that the radio option should be selected by default. That's kinda odd...

**Reply by helderneves91** [reply]
> In my case, whenever I `wire:model` an input (checkbox, radio, etc.), in my Livewire component I set the property to my desired default/starting value (if no default, then set to ''). I find using both `wire:model` and `:selected` to be confusing.

The value should be a string not an integer, that's the problem... 🫠

**Reply by jeffchown** [reply]
Yes, that does look like the main problem - Re: trying to `wire:model` an integer property for a radio group, this has been reported in issue https://github.com/livewire/flux/issues/73
